documentation/OtherMLPackages.md

```diff
@@ -0,0 +1,62 @@
+Specifically, so long as there's some sort of 'estimator' concept which provides `fit()` and `predict()` methods (the former with a `sample_weight=` argument), then those estimators can be used with our mitigation algorithms.
+However, we do not currently have tests proving that we do offer this sort of compatibility with a variety of popular ML packages.
+
+## Target Packages
+
+We should show compatibility with the following ML packages:
+  - XGBoost
+  - LightGBM
+  - TensorFlow
+  - PyTorch
+
+This is not an exhaustive list, but is a good starting point.
+In all cases, these packages either offer a SciKit-Learn compatible API, or there is a SciKit-Learn compatible wrapper available (such as `skorch` for PyTorch).
+
+## Required Tests
+
+The primary goal of this testing is to show that other ML packages can be used with our mitigation algorithms.
+This does not require in-depth testing, but more a smoke test.
+The tests should verify that, with a reasonable dataset, estimators from each package (possibly only a single example estimator) can be used with our mitigation algorithms, and produce reasonable results.
+A sample test might look a lot like those in `test_exponentiatedgradient_smoke.py` but on a more realistic dataset, and with fewer combinations of constraints and error bounds.
+
+## Concerns for Testing
+
+The packages listed above could be fairly described as 'not lightweight.'
+Requiring developers to install all of them as part of the standard developer setup (i.e. listing in `requirements-dev.txt`) will introduce friction to the process.
+Furthermore I am concerned that
+  - Some of these packages might have incompatible dependencies
+  - Some might be best installed with `conda` rather than `pip`
+
+Even if there aren't incompatible dependencies, moving to requiring `conda` for development work would be a significant change from the current `pip`-only setup (even if most developers probably `pip install` into a fresh `conda` environment).
+
+## Proposed Setup
+
+Because they will have extra dependencies, these tests should not go into our current `tests/` directory.
+We instead have a new directory `othermlpackages/` at the top level and then arrange it along these lines:
+```
+othermlpackages/
+    lightgbm/
+        conda-env.yml
+        test_lightgbm_expgrad.py
+        test_lightgbm_gridsearch.py
+    pytorch
+        conda-env.yml
+        test_pytorch_expgrad.py
+        test_pytorch_gridsearch.py
+```
+There would be new build pipelines which would run these tests.
+The tests for each package would be run in a separate job (so that they can be isolated), and the pipeline would:
+  1. Create a `conda` environment using the appropriate `conda-env.yml`
+  1. Install Fairlearn and its prerequisites into that environment
+  1. Run the tests in the appropriate directory
+
+Whether there would be one pipeline to run all the jobs, or whether we would have one pipeline for each ML package we're testing would have to be decided.
```

Comment:
I like the idea of keeping the tests separate to avoid collisions, but there's no reason you can't put these all in one automation pipeline with multiple stages. I imagine you'll want to base this on how often these kinds of failures pop up. If it's often you might want separate pipelines to make investigation easier, however my best guess is that large training packages like these are fairly stable.

Reply:
I'd say multiple jobs, rather than multiple stages (there's no sequencing going on), but I agree. If we're consistent in the layout and naming of the files, then the same basic template should be usable between all the difference package pipelines, and we can run them as multiple jobs in one pipeline or in separate pipeline as appropriate.

Then we can do both - have the 'reliable' packages in a single pipeline, and the 'unreliable' ones in their own pipelines.